### PR TITLE
Update cloud-provider images

### PR DESCRIPTION
**Pull request by hebelsan**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform aws

**What this PR does / why we need it**:
Update cloud provider images (including CSI driver images)

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Update cloud-provider images including CSI driver images
```

### CODE REVIEW

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -234,7 +238,7 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				MaxUnavailable: worker.DistributePositiveIntOrPercent(zoneIdx, pool.MaxUnavailable, zoneLen, pool.Minimum),
 				// TODO: remove the csi topology label when AWS CSI driver stops using the aws csi topology key - https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/899
 				// add aws csi driver topology label if it's not specified
-				Labels:                       utils.MergeStringMaps(pool.Labels, map[string]string{awsCSIDriverTopologyKey: zone}),
+				Labels:                       utils.MergeStringMaps(pool.Labels, map[string]string{AwsCSIDriverTopologyKey: zone}),
```
Comment (kon-angelo) [inline pkg/controller/worker/machines.go:241]: Can you remove the comment ? I don't think this is applicable, although we have to use a topology label ourselves still.

File: pkg/controller/worker/machines.go [hunk 2]
```diff
@@ -32,6 +32,11 @@ import (
 	awsapihelper "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/helper"
 )
 
+const (
+	// AwsCSIDriverTopologyKey describes the topology key used by the AWS CSI driver
+	AwsCSIDriverTopologyKey = "topology.kubernetes.io/zone"
```
Comment (kon-angelo) [inline pkg/controller/worker/machines.go:37]: Can you maybe add a comment with the issue/PR for the upstream change ? I think for the near future, there may be questions about this.

Also I would remove `Aws` prefix

File: imagevector/images.yaml [hunk 3]
```diff
@@ -223,7 +223,7 @@ images:
 - name: csi-snapshot-validation-webhook
   sourceRepository: github.com/kubernetes-csi/external-snapshotter
   repository: registry.k8s.io/sig-storage/snapshot-validation-webhook
-  tag: "v7.0.1"
+  tag: "v8.1.0"
```
Comment (kon-angelo) [inline imagevector/images.yaml:226]: Did you test the snapshot validation with these changes ? I believe that there are some RBAC changes that we did in the other upgrade PRs (e.g. https://github.com/gardener/gardener-extension-provider-azure/pull/974)

**Review by kon-angelo — CHANGES_REQUESTED** [review]
Did you try to make an "upgrade" test to check if the new EBS CSI version created any compatibility issue ? From the upgrade notes that shouldn't be the case, but better be safe

File: imagevector/images.yaml (hunk 3 above)
Comment (hebelsan) [reply imagevector/images.yaml:240]: Done

**Comment by hebelsan** [comment]
Locally tested PV upgrade and snapshot creation.

File: charts/internal/shoot-system-components/charts/csi-driver-node/templates/clusterrole-csi-provisioner.yaml [hunk 4]
```diff
@@ -6,10 +6,10 @@ metadata:
 rules:
 - apiGroups: [""]
   resources: ["persistentvolumes"]
-  verbs: ["get", "list", "watch", "create", "delete"]
+  verbs: ["get", "list", "watch", "create", "patch", "delete"]
```
Comment (kon-angelo) [inline charts/internal/shoot-system-components/charts/csi-driver-node/templates/clusterrole-csi-provisioner.yaml:9]: Can you check the other repos too for this change ?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]

**Pull request merged**